exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1349 @@
+package kucoin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/account"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/deposit"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/protocol"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// GetDefaultConfig returns a default exchange config
+func (ku *Kucoin) GetDefaultConfig() (*config.Exchange, error) {
+	ku.SetDefaults()
+	exchCfg := new(config.Exchange)
+	exchCfg.Name = ku.Name
+	exchCfg.HTTPTimeout = exchange.DefaultHTTPTimeout
+	exchCfg.BaseCurrencies = ku.BaseCurrencies
+
+	if ku.Features.Supports.RESTCapabilities.AutoPairUpdates {
+		err := ku.UpdateTradablePairs(context.TODO(), true)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return exchCfg, nil
+}
+
+// SetDefaults sets the basic defaults for Kucoin
+func (ku *Kucoin) SetDefaults() {
+	ku.Name = "Kucoin"
+	ku.Enabled = true
+	ku.Verbose = false
+
+	ku.API.CredentialsValidator.RequiresKey = true
+	ku.API.CredentialsValidator.RequiresSecret = true
+	ku.API.CredentialsValidator.RequiresPEM = true
```

Comment:
Typically, when an exchange requires a passphrase with the API key, we use `RequiresClientID`